docs/reference/adaptors.rst

```diff
@@ -383,6 +383,40 @@ Adaptors
         * :func:`flux::scan`
         * :func:`flux::fold_first`
 
+``set_union``
+^^^^^^^^^^^^^
+
+..  function::
+    template <sequence Seq1, sequence Seq2, typename Cmp = std::ranges::less> \
+        requires compatible<Seq1, Seq2> and \
+                 strict_weak_order_for<Cmp, Seq1> and \
+                 strict_weak_order_for<Cmp, Seq2> \
+    auto set_union(Seq1&& seq1, Seq2&& seq2, Cmp cmp = {}) -> sequence auto;
+
+    Returns a sequence adaptor which yields the set union of the two input sequences :var:`seq1` and :var:`seq2`, ordered by the given comparison function :var:`cmp`.
+
+    This function assumes that both :var:`seq1` and :var:`seq2` are sorted in ascending order with respect to the comparison function :var:`cmp`. If the input sequences are not sorted, the behavior is undefined.
```

Comment:
I'd prefer not to say "behaviour is undefined", because it makes people think of language UB, where the compiler can do crazy things (and which Flux tries very hard to avoid!).

The best suggestion I have for an alternative is something like

> "If the input sequences are not sorted, the contents of the resulting sequence is unspecified"

which is a bit less scary.

What do you think?

Reply:
I agree. Fixed.